AWSAuthSDK/Sources/AWSMobileClient/AWSMobileClient.swift

```diff
@@ -274,32 +274,33 @@ final public class AWSMobileClient: _AWSMobileClient {
                 self.update(self)
                 self.internalCredentialsProvider?.setIdentityProviderManagerOnce(self)
                 self.registerConfigSignInProviders()
-                
-                if (self.internalCredentialsProvider?.identityId != nil) {
-                    if (federationProvider == .none) {
-                        currentUserState = .guest
-                        completionHandler(.guest, nil)
-                    } else {
-                        currentUserState = .signedIn
-                        completionHandler(.signedIn, nil)
-                    }
-                } else  if (self.cachedLoginsMap.count > 0) {
-                    currentUserState = .signedIn
-                    completionHandler(.signedIn, nil)
-                } else {
-                    currentUserState = .signedOut
-                    completionHandler(.signedOut, nil)
-                }
-            } else if self.cachedLoginsMap.count > 0 {
-                currentUserState = .signedIn
-                completionHandler(.signedIn, nil)
-            } else {
+
+            }
+            let userState = determineIntialUserState()
+            if userState == .signedIn
+                && (federationProvider == .userPools || federationProvider == .hostedUI)
+                && self.username == nil {
+                self.signOut()
                 currentUserState = .signedOut
-                completionHandler(.signedOut, nil)
+            } else {
+                currentUserState = userState
             }
+            completionHandler(userState, nil)
```

Comment:
Should the completion handler be passed userState or currentUserState?

Reply:
Thanks for pointing it out, fixed.